drivers/i3c/i3c_renesas_imx3112.c

```diff
@@ -0,0 +1,416 @@
+				ARRAY_SIZE(i3c_renesas_imx3112_dev_arr_##parent_inst##_##ch_num),  \
+	};                                                                                         \
+	DEVICE_DT_DEFINE(node_id, i3c_renesas_imx3112_channel_initialize, NULL, NULL,              \
+			 &imx3112_channel_##parent_inst##_##ch_num##_config, POST_KERNEL,          \
+			 CONFIG_I3C_RENESAS_CHANNEL_PRIORITY, &imx3112_api_funcs);
+
+#define CHECK_MUX_CONFIG(n)									   \
+	COND_CODE_1(DT_INST_PROP_OR(n, has_external_resistor, false),                              \
+		    (BUILD_ASSERT(DT_INST_ENUM_IDX_OR(n, io_voltage, 0) < 3)), ())
+
+#define I3C_DEVICE_INIT_RENESAS_IMX3112(n)                                                         \
+	/* Internal resistors only regulate up to the VDD supply */                                \
+	CHECK_MUX_CONFIG(n)									   \
+	static const struct i3c_renesas_imx3112_mux_config i3c_renesas_imx3112_mux_config_##n = {  \
+		.bus = DEVICE_DT_GET(DT_INST_BUS(n)),                                              \
+		.dev_id = I3C_DEVICE_ID_DT_INST(n),                                                \
+		.hw_config.has_external_resistor =                                                 \
+			DT_INST_PROP_OR(n, has_external_resistor, false),                          \
+		.hw_config.io_voltage = DT_INST_ENUM_IDX_OR(n, io_voltage, 0),                     \
+		.hw_config.ch_resistance[0].scl =                                                  \
+			DT_ENUM_IDX_OR(DT_INST_CHILD(n, mux_i3c_0), scl_resistance, 0),            \
+		.hw_config.ch_resistance[0].sda =                                                  \
+			DT_ENUM_IDX_OR(DT_INST_CHILD(n, mux_i3c_0), sda_resistance, 0),            \
+		.hw_config.ch_resistance[1].scl =                                                  \
+			DT_ENUM_IDX_OR(DT_INST_CHILD(n, mux_i3c_1), scl_resistance, 0),            \
+		.hw_config.ch_resistance[1].sda =                                                  \
+			DT_ENUM_IDX_OR(DT_INST_CHILD(n, mux_i3c_1), sda_resistance, 0),            \
+	};                                                                                         \
+	static struct i3c_renesas_imx3112_mux_data i3c_renesas_imx3112_mux_data_##n = {            \
+		.lock = Z_MUTEX_INITIALIZER(i3c_renesas_imx3112_mux_data_##n.lock),                \
+		.mux_state = {.enabled = false, .channel_mask = 0},                                \
+	};                                                                                         \
+	DEVICE_DT_INST_DEFINE(n, i3c_renesas_imx3112_mux_initialize, NULL,                         \
+			      &i3c_renesas_imx3112_mux_data_##n,                                   \
+			      &i3c_renesas_imx3112_mux_config_##n, POST_KERNEL,                    \
+			      CONFIG_I3C_RENESAS_MUX_PRIORITY, NULL);                              \
+	COND_CODE_1(DT_NODE_HAS_STATUS(DT_INST_CHILD(n, mux_i3c_0), okay),                         \
```

Comment:
Would DT_INST_FOREACH() an option here? I'm not sure, do 1:4 and 1:8 versions of this exist that might justify a more generic driver?

Reply:
There are no 1:4/1:8 versions of this mux. There are 2:4/2:8 intelligent switches for I3C but those have a completely different API, so it wouldn't really make sense to try to generalize imho